### PR TITLE
WS notification/reconnection

### PR DESCRIPTION
**Pull request by pkoryzna**

### CODE REVIEW

File: frontend/src/main/scala/akka/viz/frontend/FrontendApp.scala [hunk 1]
```diff
@@ -123,38 +126,62 @@ object FrontendApp extends JSApp with Persistence
   val messagesPanel = new MessagesPanel(selectedActors)
   val userIsEnabled = Var(false)
   val onOffPanel = new OnOffPanel(_eventsEnabled, userIsEnabled)
+  val connectionAlert = new Alert()
 
   @JSExport("toggleActor")
   def toggleActor(name: String) = actorSelector.toggleActor(name)
```
Comment (lustefaniak) [inline frontend/src/main/scala/akka/viz/frontend/FrontendApp.scala]: Codacy Issue found: [Method is longer than 50 lines](https://www.codacy.com/app/lustefaniak/akka-viz/file/2050919478/issues/source?bid=3121282&fileBranchId=3133751#l134)

File: frontend/src/main/scala/akka/viz/frontend/ApiConnection.scala [hunk 2]
```diff
@@ -1,44 +1,47 @@
 package akka.viz.frontend
 
-import org.scalajs.dom._
-import org.scalajs.dom.raw.WebSocket
+import org.scalajs.dom
+import org.scalajs.dom.raw._
 
-object ApiConnection {
+import scala.concurrent.Promise
```
Comment (lustefaniak) [inline frontend/src/main/scala/akka/viz/frontend/ApiConnection.scala]: Codacy Issue found: [Use brackets to group imports from the same package](https://www.codacy.com/app/lustefaniak/akka-viz/file/2050919475/issues/source?bid=3121282&fileBranchId=3133751#l6)

File: frontend/src/main/scala/akka/viz/frontend/ApiConnection.scala [hunk 3]
```diff
@@ -1,44 +1,47 @@
...
-        console.log(e)
-        webSocket = None
-      }
-      webSocket = Some(ws)
       ws
     }
-    webSocket = Some(createWebsocket)
 
-    def getWs = webSocket.getOrElse(createWebsocket)
+    val wsPromise = Promise[WebSocket]()
```
Comment (lustefaniak) [inline frontend/src/main/scala/akka/viz/frontend/ApiConnection.scala]: Codacy Issue found: [Avoid creating Promises directly.](https://www.codacy.com/app/lustefaniak/akka-viz/file/2050919475/issues/source?bid=3121282&fileBranchId=3133751#l25)

Comment (pkoryzna) [inline frontend/src/main/scala/akka/viz/frontend/ApiConnection.scala]: WebSocket is callback-based, the only other option would be a busy-wait on WS inside a Future (bad idea in single-threaded code)

File: frontend/src/main/scala/akka/viz/frontend/FrontendApp.scala [hunk 4]
```diff
@@ -123,38 +127,69 @@ object FrontendApp extends JSApp with Persistence
...
-      upstream.send(write(SetAllowedMessages(selectedMessages.now)))
-    }
+  val maxRetries = 10
 
-    selectedActors.triggerLater {
-      console.log(s"Will send ObserveActors: ${selectedActors.now.mkString("[", ",", "]")}")
-      import upickle.default._
-      upstream.send(write(ObserveActors(selectedActors.now)))
-    }
+  def main(): Unit = {
```
Comment (lustefaniak) [inline frontend/src/main/scala/akka/viz/frontend/FrontendApp.scala]: Codacy Issue found: [Method is longer than 50 lines](https://www.codacy.com/app/lustefaniak/akka-viz/file/2051036535/issues/source?bid=3121282&fileBranchId=3133751#l137)

File: frontend/src/main/scala/akka/viz/frontend/ApiConnection.scala (hunk 2 above)
Comment (lustefaniak) [inline frontend/src/main/scala/akka/viz/frontend/ApiConnection.scala]: Codacy Issue found: [Use brackets to group imports from the same package](https://www.codacy.com/app/lustefaniak/akka-viz/file/2051036532/issues/source?bid=3121282&fileBranchId=3133751#l6)

File: frontend/src/main/scala/akka/viz/frontend/ApiConnection.scala (hunk 3 above)
Comment (lustefaniak) [inline frontend/src/main/scala/akka/viz/frontend/ApiConnection.scala]: Codacy Issue found: [Avoid creating Promises directly.](https://www.codacy.com/app/lustefaniak/akka-viz/file/2051036532/issues/source?bid=3121282&fileBranchId=3133751#l25)

Comment (pkoryzna) [inline frontend/src/main/scala/akka/viz/frontend/ApiConnection.scala]: WebSocket is callback-based, the only other option would be a busy-wait on WS inside a Future (bad idea in single-threaded code)

File: frontend/src/main/scala/akka/viz/frontend/FrontendApp.scala (hunk 4 above)
Comment (lustefaniak) [inline frontend/src/main/scala/akka/viz/frontend/FrontendApp.scala]: Codacy Issue found: [Method is longer than 50 lines](https://www.codacy.com/app/lustefaniak/akka-viz/file/2051092141/issues/source?bid=3121282&fileBranchId=3133751#l137)

File: frontend/src/main/scala/akka/viz/frontend/ApiConnection.scala (hunk 2 above)
Comment (lustefaniak) [inline frontend/src/main/scala/akka/viz/frontend/ApiConnection.scala]: Codacy Issue found: [Use brackets to group imports from the same package](https://www.codacy.com/app/lustefaniak/akka-viz/file/2051092138/issues/source?bid=3121282&fileBranchId=3133751#l6)

File: frontend/src/main/scala/akka/viz/frontend/ApiConnection.scala (hunk 3 above)
Comment (lustefaniak) [inline frontend/src/main/scala/akka/viz/frontend/ApiConnection.scala]: Codacy Issue found: [Avoid creating Promises directly.](https://www.codacy.com/app/lustefaniak/akka-viz/file/2051092138/issues/source?bid=3121282&fileBranchId=3133751#l25)

**Comment by JJag** [comment]
:+1:

**Pull request merged**